airbyte-integrations/connectors/source-zoom/source_zoom/components.py

```diff
@@ -0,0 +1,64 @@
+import base64
+import requests
+import time
+
+from dataclasses import dataclass
+from http import HTTPStatus
+from typing import Any, Mapping, Union
+
+from airbyte_cdk.sources.declarative.auth.declarative_authenticator import NoAuth
+from airbyte_cdk.sources.declarative.interpolation import InterpolatedString
+from airbyte_cdk.sources.declarative.types import Config
+from requests import HTTPError
+
+# https://developers.zoom.us/docs/internal-apps/s2s-oauth/#successful-response
+# The Bearer token generated by server-to-server token will expire in one hour
+BEARER_TOKEN_EXPIRES_IN = 3590
+
+
+@dataclass
+class ServerToServerOauthAuthenticator(NoAuth):
+    config: Config
+    account_id: Union[InterpolatedString, str]
+    client_id: Union[InterpolatedString, str]
+    client_secret: Union[InterpolatedString, str]
+    grant_type: Union[InterpolatedString, str]
+    authorization_endpoint: Union[InterpolatedString, str]
+
+    _generate_token_time = 0
```

Comment:
@KqLLL add unit test to validate the functions inside the class

Reply:
Added unit tests